crates/sui-graphql-rpc/src/server/data_provider.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// For testing, use existing RPC as data source
+
+use std::str::FromStr;
+
+use crate::types::address::Address;
+use crate::types::balance::Balance;
+use crate::types::base64::Base64;
+use crate::types::big_int::BigInt;
+use crate::types::transaction_block::TransactionBlock;
+use crate::types::{object::Object, sui_address::SuiAddress};
+use async_graphql::*;
+use sui_json_rpc_types::{
+    SuiObjectDataOptions, SuiPastObjectResponse, SuiTransactionBlockResponseOptions,
+};
+use sui_sdk::types::base_types::ObjectID;
+use sui_sdk::types::digests::TransactionDigest;
+use sui_sdk::SuiClient;
+
+pub(crate) async fn fetch_obj(
+    cl: &SuiClient,
+    address: SuiAddress,
+    version: Option<u64>,
+) -> Result<Option<Object>> {
+    let oid: ObjectID = address.to_array().as_slice().try_into()?;
```

Comment:
thoughts on implementing From<SuiAddress> to ObjectID? It looks like we have a few types keyed on SuiAddress, but we query them with ObjectID

Reply:
I dont want to leak sui types in too many places. But yeah I can impl helper functions in this file

Reply:
Added some conversion functions here https://github.com/MystenLabs/sui/pull/13377